ed25519/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+extern crate ed25519_dalek;
+extern crate ewasm_api;
+extern crate sha2;
+
+mod verify;
+
+#[cfg(not(test))]
+#[no_mangle]
+pub extern "C" fn main() {
+    let length = ewasm_api::calldata_size();
+
+    ewasm_api::consume_gas(2000);
+
+    if length < 128 {
```

Comment:
What is supposed to happen if length ends up being greater than 128 for whatever reason?

Reply:
Ignored. Have a look at the EIP.